### PR TITLE
feature: added variable shift to Shift

### PR DESCRIPTION
**Pull request by mrava87**
This PR modified the `Shift` operator to allow user to pass shifts that vary across all axes apart from the one the shift
is applied. An example of how to use this new feature is also provided inn the `plot_shift` example

### CODE REVIEW

File: pylops/signalprocessing/Shift.py [hunk 1]
```diff
@@ -95,8 +98,24 @@ def Shift(
...
+
+    shift = _value_or_list_like_to_array(shift)
+
+    if shift.size == 1:
+        shift = np.exp(-1j * 2 * np.pi * Fop.f * shift)
+        Sop = Diagonal(shift, dims=dimsdiag, axis=axis, dtype=Fop.cdtype)
+    else:
+        # add dimensions to shift to match dimensions of model and data
+        if axis < 0:
+            axis = len(dims) + axis
```
Comment (cako) [inline pylops/signalprocessing/Shift.py:110]: Consider using `normalize_axis_index` found in `numpy.core.multiarray`

**Review by cako — COMMENTED** [review]
Looks good!

File: pylops/signalprocessing/Shift.py (hunk 1 above)
Comment (mrava87) [reply pylops/signalprocessing/Shift.py]: Good point, added :)

**Comment by mrava87** [comment]
Thanks for the suggestion, I always forget about `normalize_axis_index`. Feel free to merge if you don't have any further comment

**Review by cako — APPROVED** [review]

**Pull request merged**